spring-cloud-sleuth-otel/src/main/java/org/springframework/cloud/sleuth/otel/bridge/OtelHttpClientHandler.java

```diff
@@ -39,156 +34,101 @@
 import org.springframework.cloud.sleuth.http.HttpRequestParser;
 import org.springframework.cloud.sleuth.http.HttpResponseParser;
 import org.springframework.lang.Nullable;
-import org.springframework.util.StringUtils;
 
 /**
  * OpenTelemetry implementation of a {@link HttpClientHandler}.
  *
  * @author Marcin Grzejszczak
+ * @author Nikita Salnikov-Tarnovski
  * @since 1.0.0
  */
-public class OtelHttpClientHandler extends HttpClientTracer<HttpClientRequest, HttpClientRequest, HttpClientResponse>
-		implements HttpClientHandler {
+public class OtelHttpClientHandler implements HttpClientHandler {
 
 	private static final Log log = LogFactory.getLog(OtelHttpClientHandler.class);
 
+	private static final ContextKey<HttpClientRequest> REQUEST_CONTEXT_KEY = ContextKey
+			.named(OtelHttpClientHandler.class.getName() + ".request");
+
 	private final HttpRequestParser httpClientRequestParser;
 
 	private final HttpResponseParser httpClientResponseParser;
 
 	private final SamplerFunction<HttpRequest> samplerFunction;
 
+	private final Instrumenter<HttpClientRequest, HttpClientResponse> instrumenter;
+
 	public OtelHttpClientHandler(OpenTelemetry openTelemetry, @Nullable HttpRequestParser httpClientRequestParser,
 			@Nullable HttpResponseParser httpClientResponseParser, SamplerFunction<HttpRequest> samplerFunction) {
-		super(openTelemetry, new NetPeerAttributes());
 		this.httpClientRequestParser = httpClientRequestParser;
 		this.httpClientResponseParser = httpClientResponseParser;
 		this.samplerFunction = samplerFunction;
-	}
 
-	@Override
-	public Context startSpan(Context parentContext, HttpClientRequest request, HttpClientRequest carrier,
-			long startTimeNanos) {
-		Context context = super.startSpan(parentContext, request, carrier, startTimeNanos);
-		io.opentelemetry.api.trace.Span span = io.opentelemetry.api.trace.Span.fromContext(context);
-		if (this.httpClientRequestParser != null) {
-			Span fromOtel = OtelSpan.fromOtel(span);
-			this.httpClientRequestParser.parse(request, fromOtel.context(), fromOtel);
-		}
-		String path = request.path();
-		if (path != null) {
-			span.setAttribute(SemanticAttributes.HTTP_ROUTE, path);
-		}
-		return context;
+		SpringHttpClientAttributesExtractor httpAttributesExtractor = new SpringHttpClientAttributesExtractor();
+		this.instrumenter = Instrumenter
+				.<HttpClientRequest, HttpClientResponse>newBuilder(openTelemetry, "org.springframework.cloud.sleuth",
+						HttpSpanNameExtractor.create(httpAttributesExtractor))
+				.addAttributesExtractor(new HttpRequestNetClientAttributesExtractor())
```

Comment:
Should we set the status extractor too? (also in server handler)
```suggestion
				.setSpanStatusExtractor(HttpSpanStatusExtractor.create(httpAttributesExtractor))
				.addAttributesExtractor(new HttpRequestNetClientAttributesExtractor())
```